umbra-js/src/utils/utils.ts

```diff
@@ -245,7 +245,12 @@ export async function getBlockNumberUserRegistered(address: string, provider: St
   address = getAddress(address); // address input validation
   const registry = new StealthKeyRegistry(provider);
   const filter = registry._registry.filters.StealthKeyChanged(address, null, null, null, null);
-  const stealthKeyLogs = await registry._registry.queryFilter(filter);
+  let stealthKeyLogs = await registry._registry.queryFilter(filter);
+  if (stealthKeyLogs.length > 1) {
+    stealthKeyLogs = stealthKeyLogs.sort(function (a, b) {
```

Comment:
We should add a test for this method, or if we don't want the RPC request during tests, pull the sorting into a separate method and unit test that

Reply:
I would pull the sort method out into something that can be unit tested, then I would call sort regardless of the length of the logs. It should be extremely fast and it's not worth short circuiting.

Reply:
Added test https://github.com/ScopeLift/umbra-protocol/pull/567/commits/b70243b6b34b90fa1aa8ac95ddbf26e1affa8902
Removed `if` checking https://github.com/ScopeLift/umbra-protocol/pull/567/commits/b3957ac615b891271ae6520ef630aeab3aa9a730